### PR TITLE
fix: use isinstance check instead of type

### PR DESCRIPTION
**Pull request by jlhood**
*Issue #, if available:* #1410

Note, had to run `make black` to correct formatting errors elsewhere in the project in order for `make pr` to pass.

*Checklist:*

- [n/a] Write/update tests
- [ x ] `make pr` passes
- [n/a] Update documentation
- [n/a] Verify transformed template deploys and application functions as expected
- [n/a] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/translator/translator.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def _get_function_names(self, resource_dict, intrinsics_resolver):
                     # adds to the function_names dict with key as the api_name and value as the function_name
                     if item.get("Type") == "Api" and item.get("Properties") and item.get("Properties").get("RestApiId"):
                         rest_api = item.get("Properties").get("RestApiId")
-                        if type(rest_api) == dict or isinstance(rest_api, OrderedDict):
+                        if isinstance(rest_api, (dict, OrderedDict)):
```
Comment (jlhood) [inline samtranslator/translator/translator.py]: This is the only non-formatting change

Comment (keetonian) [reply samtranslator/translator/translator.py]: Good, this needed to happen.

**Comment by keetonian** [comment]
Closed & reopened PR to kick off new travis run.

File: samtranslator/translator/translator.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def _get_function_names(self, resource_dict, intrinsics_resolver):
                     # adds to the function_names dict with key as the api_name and value as the function_name
                     if item.get("Type") == "Api" and item.get("Properties") and item.get("Properties").get("RestApiId"):
                         rest_api = item.get("Properties").get("RestApiId")
-                        if type(rest_api) == dict or isinstance(rest_api, dict):
+                        if isinstance(rest_api, dict):
```
Comment (keetonian) [inline samtranslator/translator/translator.py]: Resolved conflicts. I also updated this section- it turns out that an `OrderedDict` is an instance of `dict`, so this works for both cases we wanted to cover.

Comment (jlhood) [reply samtranslator/translator/translator.py]: Great, appreciate it!

**Comment by kddejong** [comment]
https://github.com/aws-cloudformation/cfn-python-lint/issues/1325

**Comment by keetonian** [comment]
@kddejong thanks for bringing this up. I will dig into this.

**Comment by kddejong** [comment]
Thanks @keetonian and @shwetaskatdare

**Pull request merged**